src/en/comichubfree/src/eu/kanade/tachiyomi/extension/en/comichubfree/ComicHubFree.kt

```diff
@@ -0,0 +1,145 @@
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class ComicHubFree : Source, ParsedHttpSource() {
+    override val baseUrl = "https://comichubfree.com"
+
+    override val lang = "en"
+
+    override val name = "ComicHubFree"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient
+
+    override fun popularMangaSelector() = ".movie-list-index > .cartoon-box"
+
+    override fun latestUpdatesSelector() = popularMangaSelector()
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun popularMangaNextPageSelector() = "ul.pagination > li:last-child"
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun chapterListSelector() = "div.episode-list > div > table > tbody > tr"
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/popular-comic?page=$page")
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/new-comic?page=$page")
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return GET("$baseUrl/search-comic?key=$query&page=$page")
+    }
+
+    override fun pageListRequest(chapter: SChapter): Request {
+        return GET(baseUrl + chapter.url + "/all")
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga {
+        return SManga.create().apply {
+            setUrlWithoutDomain(element.child(0).attr("abs:href"))
+            title = element.selectFirst("h3")!!.text()
+            val image = element.selectFirst("img")!!
+            thumbnail_url = when {
+                image.hasAttr("data-src") -> image.attr("abs:data-src")
+                else -> image.attr("abs:src")
+            }
+        }
+    }
+
+    override fun latestUpdatesFromElement(element: Element): SManga {
+        return popularMangaFromElement(element)
+    }
+
+    override fun searchMangaFromElement(element: Element): SManga {
+        return popularMangaFromElement(element)
+    }
+
+    override fun chapterFromElement(element: Element): SChapter {
+        val urlElement = element.select("a").first()!!
+        val dateElement = element.select("td").last()!!
+
+        val chapter = SChapter.create()
+        chapter.name = urlElement.text()
+        chapter.setUrlWithoutDomain(urlElement.attr("href"))
+        chapter.date_upload = dateElement.text().let {
+            SimpleDateFormat("d-MMM-yyyy", Locale.getDefault()).parse(it)?.time ?: 0L
+        }
```

Comment:
save dateFormat in class to avoide recreating it everytime

also:
```suggestion
        chapter.date_upload =  dateFormat.tryParse(dateElement.text())
```
need to import